keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -156,12 +155,13 @@ class ResourceActuator(
               }
           }
           else -> {
-            log.info("Resource {} is valid [$coid]", id)
-            // TODO: not sure this logic belongs here
             val lastEvent = resourceRepository.lastEvent(id)
-            if (lastEvent is ResourceDeltaDetected || lastEvent is ResourceActuationLaunched) {
+            if (lastEvent is ResourceDeltaDetected || lastEvent is ResourceActuationLaunched ||
+              lastEvent is ResourceTaskSucceeded) {
```

Comment:
I don't think that a task succeeding means a resource delta is resolved - for example, a bake task can succeed but definitely not resolve a delta.

Reply:
oh wait, this is in the section where we've already determined the resource is valid, right?

Reply:
Its a bit tricky - as for my understanding, this is the only place that we are publishing the `ResourceDeltaResolved` event. This is being determined only after checking if the resource should be created/ updated, so I guess the answer to your question is yes. My only fear with this solution is that we might send duplicated `ResourceDeltaResolved`, so I'm wondering if there's some logic in the UI to handle it (or should we handle it here in some way?)